.github/workflows/integration-tests.yml

```
@@ -69,9 +69,20 @@ jobs:
+      - name: Restore pytest cache
+        uses: actions/cache@v3
+        with:
+          path: .pytest_cache
+          key: pytest-cache-${{ github.ref }}-${{ github.run_id }}
+          restore-keys: pytest-cache-${{ github.ref }}
       - name: Run integration tests - multiple peers - pytest
-        run: poetry -C tests run pytest tests/consensus_tests --durations=10
+        run: poetry -C tests run pytest tests/consensus_tests --failed-first --new-first --durations=10
         timeout-minutes: 60
+      - name: Save pytest cache
+        uses: actions/cache@v3
```

actionlint (1.7.4)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)